ssl/statem/statem_srvr.c

```diff
@@ -1716,7 +1743,11 @@ static int tls_early_post_process_client_hello(SSL *s, int *pal)
         }
     }
 
-    if (!s->hit && s->version >= TLS1_VERSION && s->ext.session_secret_cb) {
+    if (!s->hit
+            && s->version >= TLS1_VERSION
+            && !SSL_IS_TLS13(s)
+            && !SSL_IS_DTLS(s)
+            && s->ext.session_secret_cb) {
```

Comment:
Yes, it should be documented at some point. FYI, this primarily exists for EAP-FAST support I believe (which is a fairly obscure corner of libssl).